src/ec/eddsa.rs

```diff
@@ -297,60 +273,80 @@ fn elem_sq_in_place(a: &mut Elem) {
 #[doc(hidden)]
 #[no_mangle]
 pub unsafe extern fn eddsa_elem_invert(out: &mut Elem, a: &Elem) {
+    // Sets |out| to be |a| * |b| (mod p).
```

Comment:
Let's use the Rust style \`a\` instead of BoringSSL style |a|` in the comments.

More generally, we should make this more consistent with the P-256/P-384 code. In particular:

* In the P-256/P-384 code we usually to return a `Elem` instead of having an `Elem` out parameter. If that is less efficient then we should switch that code to use the style you used style. However, if it is just as efficient we should do things the way the P-256/P-384 code does.

* I am not wedded to the names I used in the P-256 and P-384 code but it would be good to use the same names here and there. For example, we have `elem_product` in P-256/P-384 which is like your `mul` and we have `elem_mul` which is like your `mul_in_place`. I would suggest that, given more experience with Rust convention sand things like the `MulAssign` trait, `elem_mul_assign` seems better than `elem_mul` or `mul_in_place`. (Note that here you could actually implement `Mul`, `MulAssign`, etc., but I wouldn't bother since it'd likely be undone in the not-too-distant future.)